src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs

```diff
@@ -52,6 +52,22 @@ public FileTextLoaderOptionsProvider()
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
     public class FileTextLoader : TextLoader
     {
+        /// <summary>
+        /// Guards access to <see cref="_workspace"/> and <see cref="_lazyTextAndVersion"/>.
+        /// </summary>
+        private readonly object _gate = new object();
+
+        /// <summary>
+        /// This is the workspace used to initialize <see cref="_lazyTextAndVersion"/>.
```

Comment:
maybe call tihs _lastWorkspace or doc that it's the workspace used to initialize this.

Reply:
Already documented as such.